File: GitNexus/gitnexus/test/fixtures/lang-resolution/php-return-type/app/Services/UserService.php
<?php

namespace App\Services;

use App\Models\User;

class UserService {
    public function getUser(string $name): User {
        return new User($name);
    }

    public function processUser(): void {
        $user = $this->getUser("alice");
        $user->save();
    }
}

File: GitNexus/gitnexus/test/fixtures/lang-resolution/csharp-proj/Services/UserService.cs
using MyApp.Models;
using MyApp.Interfaces;

namespace MyApp.Services
{
    public class UserService
    {
        private readonly IRepository _repo;
        private readonly ILogger _logger;

        public void CreateUser(string name)
        {
            var user = new User();
            user.Validate();
            _repo.Save();
            _logger.Log("User created");
        }
    }
}
